packages/engine/src/gltf/GLTFComponent.tsx

```diff
@@ -83,7 +86,19 @@ const ResourceReactor = (props: { documentID: string; entity: Entity }) => {
   useEffect(() => {
     if (getComponent(props.entity, GLTFComponent).progress === 100) return
     if (!getState(GLTFDocumentState)[props.documentID]) return
-
+    const document = getState(GLTFDocumentState)[props.documentID]
+    const modelNodes = document.nodes?.filter((node) => !!node.extensions?.[ModelComponent.jsonID])
+    if (modelNodes) {
+      for (const node of modelNodes) {
+        //check if an entity exists for this node, and has a model component
+        const uuid = node.extensions![UUIDComponent.jsonID] as EntityUUID
+        if (!UUIDComponent.entitiesByUUIDState[uuid]) return
+        const entity = UUIDComponent.entitiesByUUIDState[uuid].value
+        const model = getOptionalComponent(entity, ModelComponent)
```

Comment:
It doesnt look like this effect is due to rerun if the model component is added; is that correct?